common/utils.h

```diff
@@ -0,0 +1,19 @@
+#include <algorithm>
+#include <cctype>
+#include <locale>
+
+void trim_left(std::string &str)
```

Comment:
Because these are in the header, they need to be marked inline.

Reply:
Fixed.